### PR TITLE
Update top-level .gitignore file

### PR DESCRIPTION
**Pull request by razvand**
Ignore object files. Ignore archive files. Ignore temporary and swap files.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,25 @@
...
+*.a
+*.so
+
+# Ignore archive files.
+*gz
+*bz2
+*.zip
+*.tar
+*xz
+
```
Comment (gabrielmocanu) [inline .gitignore:22]:
```suggestion
*.rar

```

File: .gitignore [hunk 2]
```diff
@@ -1,6 +1,25 @@
...
+# Ignore swap files.
+*swo
+*swp
```
Comment (gabrielmocanu) [inline .gitignore:25]:
```suggestion
*swp

# Ignore GDB files.
.gdb_history
```

**Review by gabrielmocanu — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 2 above)
Comment (razvand) [reply .gitignore:25]: @gabrielmocanu , I don't understand this suggestion. What is changed?

Comment (razvand) [reply .gitignore:25]: Oh, `.gdb_history`, got it.

**Review by gabrielmocanu — APPROVED** [review]
Looks good

**Pull request merged**